pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -250,20 +264,35 @@ func (pl *InterPodAffinity) PreFilter(ctx context.Context, cycleState *framework
 	if podInfo.ParseError != nil {
 		return framework.NewStatus(framework.UnschedulableAndUnresolvable, fmt.Sprintf("parsing pod: %+v", podInfo.ParseError))
 	}
+	podNS, err := pl.nsLister.Get(pod.Namespace)
+	if err != nil {
+		return framework.NewStatus(framework.Error, fmt.Sprintf("failed to get pod namespace object: %v", err))
+	}
+	namespaceLabels := labels.Merge(podNS.Labels, nil)
+
+	// Merge the namespaces selected by NamespaceSelector of each term.
+	for _, t := range podInfo.RequiredAffinityTerms {
```

Comment:
Since we are planing to share podInfo from the cache (#97813), we need to make sure we are cloning it here before storing it in the prefilter state.

perhaps we should not store podInfo, but just the items that we actually need from it? That would avoid confusion of why we are storing podInfo if it's being passed and prevent overrides.

Reply:
My original plan was to embed PodInfo in QueuedPodInfo. The PodInfo we store in the cache will continue to be calculated as it is right now on `add`.